### PR TITLE
Migrating to workmanager

### LINKED ISSUE #194: Introduce WorkManager
#### Problem description
We use Evernote's android-job library for scheduling the tasks but is deprecated and not supported anymore by the developers of the library. So, we should start moving towards android [WorkManager](https://developer.android.com/topic/libraries/architecture/workmanager).

#### Steps to reproduce the problem
Not an issue, code enhancement

#### Expected behavior
Should not break the current functionality, but should completely replace android-job with WorkManager

#### Other information 
Things you tried, stack traces, related issues, suggestions on how to fix it...

### PR DESCRIPTION
**Pull request by udhay24**
Closes #194 

#### What has been done to verify that this works as intended?
Removed Evernote's android job dependency and introduced the android WorkManger. The app is found to be working properly and passed Gradle build's lint , pmd, check style, findbugs tests.
the app is tested by sending and receiving the form between lenovo K53a48(android 7.0)-for sending and Samsung j7 2016(android 8.1)- for receiving.
     
#### Why is this the best possible solution? Were any other approaches considered?
Made sure the underlying algorithm is not changed and only the required components have been switch to work manager.
#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
Since the work manager needed a compiletimesdk of min 28 and androidx dependencies were used  this branch comes from #189 branch.
#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew pmd checkstyle lint findbugs and confirmed all checks still pass OR confirm CircleCI build passes
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/share/blob/master/share_app/src/main/assets/open_source_licenses.html).

### CODE REVIEW

**Comment by shobhitagarwal1612** [comment]
> What has been done to verify that this works as intended?

Please mention if you verified by sending/receiving forms form one device to another. If not, then please do test and paste your results here

File: skunkworks_crow/build.gradle [hunk 1]
```diff
@@ -55,7 +55,10 @@ dependencies {
     implementation "com.jakewharton.rxrelay2:rxrelay:2.0.0"
 
     // evernote's job scheduler
```
Comment (lakshyagupta21) [inline skunkworks_crow/build.gradle]: remove this comment as well.

**Comment by lakshyagupta21** [comment]
@udhay24 Also resolve merge conflicts.

File: skunkworks_crow/src/main/java/org/odk/share/services/SenderService.java [hunk 2]
```diff
@@ -80,22 +80,20 @@ public void startUploading(long[] instancesToSend, int port, int mode) {
...
-        request.schedule();
-        Timber.d("Starting upload job %d : ", request.getJobId());
+    private void startJob(WorkRequest request) {
+        workManager.enqueue(request);
+        Timber.d("Starting upload job %s : ", request.getId().toString());
         currentJob = request;
     }
 
     public void cancel() {
         if (currentJob != null) {
```
[the remark this answers is not in the document]
Comment (udhay24) [reply skunkworks_crow/src/main/java/org/odk/share/services/SenderService.java]: can you specify the line where i should add the indentation please?

File: skunkworks_crow/src/main/java/org/odk/share/tasks/DownloadJob.java [hunk 3]
```diff
@@ -148,8 +156,9 @@ private DownloadEvent receiveForms() {
     }
 
     @Override
-    protected void onCancel() {
-        try {
+    public void onStopped() {
```
[the remark this answers is not in the document]
Comment (udhay24) [reply skunkworks_crow/src/main/java/org/odk/share/tasks/DownloadJob.java]: added indentation at line 161

**Comment by udhay24** [comment]
> @udhay24 Also resolve merge conflicts.

shouldn't i wait until #198 is merged? because this PR depends on it for updating the target sdk to 28

File: skunkworks_crow/src/main/java/org/odk/share/services/SenderService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (lakshyagupta21) [reply skunkworks_crow/src/main/java/org/odk/share/services/SenderService.java]: Current code is 
```
public void cancel() {
        if (currentJob != null) {
                workManager.cancelWorkById(currentJob.getId());
                rxEventBus.post(new UploadEvent(UploadEvent.Status.CANCELLED));
            } else {
                Timber.e("Pending job not found : %s", currentJob);
            }
            currentJob = null;
        }
    }
```
And it should be like 

```
public void cancel() {
    if (currentJob != null) {
        workManager.cancelWorkById(currentJob.getId());
        rxEventBus.post(new UploadEvent(UploadEvent.Status.CANCELLED));
     } else {
        Timber.e("Pending job not found : %s", currentJob);
     }
     currentJob = null;
}
```

File: skunkworks_crow/src/main/java/org/odk/share/services/SenderService.java [hunk 4]
```diff
@@ -80,22 +80,20 @@ public void startUploading(long[] instancesToSend, int port, int mode) {
...
-            if (job != null) {
-                job.cancel();
+                workManager.cancelWorkById(currentJob.getId());
                 rxEventBus.post(new UploadEvent(UploadEvent.Status.CANCELLED));
             } else {
                 Timber.e("Pending job not found : %s", currentJob);
             }
             currentJob = null;
         }
     }
```
Comment (lakshyagupta21) [inline skunkworks_crow/src/main/java/org/odk/share/services/SenderService.java]: Also, as this is closing braces please indent this as well.

Comment (udhay24) [reply skunkworks_crow/src/main/java/org/odk/share/services/SenderService.java]: removed the indentation for the SenderService closing braces.

**Comment by lakshyagupta21** [comment]
@udhay24 Any updates on this? I'm seeing lot of changes and commits. Can you squash them and resolve the conflicts.

**Comment by lakshyagupta21** [comment]
@udhay24 Closing this PR as there is no update on this, open this again if required.

**Pull request closed without merging**